modules/xmpp/XmppConnection.js

```diff
@@ -260,7 +264,9 @@ export default class XmppConnection extends Listenable {
 
         if (status === Strophe.Status.CONNECTED || status === Strophe.Status.ATTACHED) {
             this._maybeEnableStreamResume();
-            this._maybeStartWSKeepAlive();
+
+            // after connecting - immediately check whether shard changed and maybe schedule keep alive
+            this._checkShardAndMaybeScheduleKeepAlive();
```

Comment:
Here call `keepAliveAndCheckShard` mentioned in another comment, also leave the comment which explains it's doing the check immediately.
Just after that leave the `this._maybeStartWSKeepAlive();` which is supposed to schedule the keep alive